Y0/y0_euler.py

```diff
@@ -0,0 +1,294 @@
+            Field[5] = Min;
+            Field[5].FieldsList = {2,4};
+
+            Background Field = 5;
+        """, ["data/pseudoY0.brep"]), 3, target_unit='MM')
+    else:
+        mesh = read_gmsh("data/pseudoY0.msh")
+
+    return mesh
+
+
+def main(ctx_factory=cl.create_some_context):
+    """Drive the Y0 example."""
+    cl_ctx = ctx_factory()
+    queue = cl.CommandQueue(cl_ctx)
+    actx = PyOpenCLArrayContext(queue)
+
+    logger = logging.getLogger(__name__)
+
+    dim = 3
+    order = 1
+    exittol = .09
+    t_final = 0.1
+    current_cfl = 1.0
+    vel_init = np.zeros(shape=(dim,))
+    vel_inflow = np.zeros(shape=(dim,))
+    orig = np.zeros(shape=(dim,))
+    orig[0] = 0.83
+    orig[2] = 0.001
+    #    vel[0] = 340.0
+    vel_inflow[0] = 100.0  # m/s
+    current_dt = 1e-8
+    current_t = 0
+    eos = IdealSingleGas()
+    bulk_init = Lump(numdim=dim, rho0=1.225, p0=100000.0,
+                     center=orig, velocity=vel_init, rhoamp=0.0)
+    inflow_init = Lump(numdim=dim, rho0=1.225, p0=200000.0,
+                       center=orig, velocity=vel_inflow, rhoamp=0.0)
+
+    casename = "pseudoY0"
+    wall = AdiabaticSlipBoundary()
+    from grudge import sym
+    #    boundaries = {BTAG_ALL: PrescribedBoundary(initializer)}
+    boundaries = {sym.DTAG_BOUNDARY("Inflow"): PrescribedBoundary(inflow_init),
+                  sym.DTAG_BOUNDARY("Outflow"): DummyBoundary,
+                  sym.DTAG_BOUNDARY("Wall"): wall}
+    constant_cfl = False
+    nstatus = 10
+    nviz = 10
+    rank = 0
+    checkpoint_t = current_t
+    current_step = 0
+    timestepper = rk4_step
+
+    comm = MPI.COMM_WORLD
+    nproc = comm.Get_size()
+    rank = comm.Get_rank()
+    num_parts = nproc
+
+    from meshmode.distributed import (
+        MPIMeshDistributor,
+        get_partition_by_pymetis,
+    )
+
+    global_nelements = 0
+    local_nelements = 0
+
+    if nproc > 1:
+        mesh_dist = MPIMeshDistributor(comm)
+        if mesh_dist.is_mananger_rank():
+
+            mesh = get_pseudo_y0_mesh()
+            global_nelements = mesh.nelements
+            logging.info(f"Total {dim}d elements: {global_nelements}")
+            logging.info(f"Grid BTAGS: {mesh.boundary_tags}")
+            #            print(f"btags = {mesh.boundary_tags}")
+
+            logging.info("Partitioning grid.")
+            part_per_element = get_partition_by_pymetis(mesh, num_parts)
+            logging.info("Sending grid partitions")
+
+            local_mesh = mesh_dist.send_mesh_parts(mesh, part_per_element, num_parts)
+            del mesh
+            logging.info("Grid partitions sent.")
+
+        else:
+            local_mesh = mesh_dist.receive_mesh_part()
+
+        comm.Barrier()
+
+        if rank == 0:
+            logging.info("Grid distributed.")
+    else:
+        logging.info("Reading grid.")
+        local_mesh = get_pseudo_y0_mesh()
+        global_nelements = local_mesh.nelements
+        logging.info("Done. Reading grid.")
```

Comment:
Doesn't use new distributed-mesh widget.

Reply:
This is fixed... but... let's try to remember to discuss a related issue regarding merging experiments, production drivers, etc. 

Essentially, now I am worried that after I respond to all the review comments, that the code I check in is not the code that was actually used to produce the results, and it takes actual multiple days to verify that what I do get with the new code does/can/will ever match the previous results.  

Our current case is very easy - but the actual production/prediction cases have the potential to be much worse.  It's just something to put on our collective radar.